### PR TITLE
OtelTraceProvider.Builder introduce the trace rate limit property

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: features/dd-sdk-android-trace-otel/src/main/kotlin/com/datadog/android/trace/opentelemetry/OtelTracerProvider.kt [hunk 1]
```diff
@@ -236,6 +239,19 @@ class OtelTracerProvider internal constructor(
...
         }
 
+        /**
+         * Sets the trace rate limit. This is the maximum number of traces per second that will be
+         * accepted. Please not that this property is used in conjunction with the sample rate. If no sample rate
+         * is provided this property and its related logic will be ignored.
+         * @param traceRateLimit the trace rate limit as a value between 1 and Int.MAX_VALUE (default is Int.MAX_VALUE)
+         */
+        fun setTraceRateLimit(
+            @IntRange(from = 1, to = Int.MAX_VALUE.toLong()) traceRateLimit: Int
```
Comment (0xnm) [inline features/dd-sdk-android-trace-otel/src/main/kotlin/com/datadog/android/trace/opentelemetry/OtelTracerProvider.kt:249]: why min value is 1? shouldn't it be 0 instead to be consistent with sample rate? there we allow setting 0, meaning no traces will be sent.

File: reliability/single-fit/trace/src/test/kotlin/com/datadog/android/trace/integration/otel/OtelTracerProviderTest.kt [hunk 2]
```diff
@@ -769,7 +770,7 @@ internal class OtelTracerProviderTest {
             }
     }
 
-    @Test
+    @RepeatedTest(10)
```
Comment (0xnm) [inline reliability/single-fit/trace/src/test/kotlin/com/datadog/android/trace/integration/otel/OtelTracerProviderTest.kt:773]: do we have any added value from repeating this test, given that we already have quite large number of spans in the test itself?

**Review by 0xnm — APPROVED** [review]
lgtm, I've added some questions/comments

File: features/dd-sdk-android-trace-otel/src/main/kotlin/com/datadog/android/trace/opentelemetry/OtelTracerProvider.kt (hunk 1 above)
Comment (mariusc83) [reply features/dd-sdk-android-trace-otel/src/main/kotlin/com/datadog/android/trace/opentelemetry/OtelTracerProvider.kt:249]: that's what they have in the code base

File: reliability/single-fit/trace/src/test/kotlin/com/datadog/android/trace/integration/otel/OtelTracerProviderTest.kt (hunk 2 above)
Comment (mariusc83) [reply reliability/single-fit/trace/src/test/kotlin/com/datadog/android/trace/integration/otel/OtelTracerProviderTest.kt:773]: not quite...you are right, going to lower it.

File: features/dd-sdk-android-trace-otel/src/main/kotlin/com/datadog/android/trace/opentelemetry/OtelTracerProvider.kt (hunk 1 above)
Comment (mariusc83) [reply features/dd-sdk-android-trace-otel/src/main/kotlin/com/datadog/android/trace/opentelemetry/OtelTracerProvider.kt:249]: it is a `Math.max(1, traceLimit)` that it's being used in that class so `0` will not be taken into account.

**Review by jonathanmos — APPROVED** [review]

**Pull request merged**